Cabal/Distribution/Package/TextClass.hs

```diff
@@ -27,6 +27,18 @@ instance Text Dependency where
              Parse.skipSpaces
              return (Dependency name ver)
 
+instance Text ExeDependency where
+  disp (ExeDependency name exe ver) =
+    (disp name <<>> Disp.text ":" <<>> disp exe) <+> disp ver
+
+  parse = do name <- parse
+             _ <- Parse.char ':'
+             exe <- parse
```

Comment:
Besides laziness, I figured a blanket depend-on-all-the-exes dependency might complicate making the solver use deps on specific components. Then again we may need a blanket package dep anyways.